modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/filename/PdsFolderResolver.java

```diff
@@ -0,0 +1,524 @@
+
+    /**
+     * Subdir (nodeII-UID, where II - node index as decimal integer, UID - string representation of consistent ID)
+     * pattern.
+     */
+    private static final String SUBDIR_PATTERN = NODE_PATTERN + UUID_STR_PATTERN;
+
+    /** Database subfolders for new style filter. */
+    public static final FileFilter DB_SUBFOLDERS_NEW_STYLE_FILTER = new FileFilter() {
+        @Override public boolean accept(File pathname) {
+            return pathname.isDirectory() && pathname.getName().matches(SUBDIR_PATTERN);
+        }
+    };
+
+    /** Database subfolders for old style filter. */
+    private static final FileFilter DB_SUBFOLDERS_OLD_STYLE_FILTER = new FileFilter() {
+        @Override public boolean accept(File pathname) {
+            String path = pathname.toString();
+            return pathname.isDirectory()
+                && !"wal".equals(pathname.getName())
+                && !path.contains(DataStorageConfiguration.DFLT_BINARY_METADATA_PATH)
+                && !path.contains(DataStorageConfiguration.DFLT_MARSHALLER_PATH)
+                && !pathname.getName().matches(SUBDIR_PATTERN);
+        }
+    };
+
+    /** Database default folder. */
+    public static final String DB_DEFAULT_FOLDER = "db";
+
+    /** */
+    private final IgniteConfiguration cfg;
+
+    /** */
+    private final IgniteLogger log;
+
+    /** */
+    private final @Nullable Serializable consistentId;
+
+    /** */
+    private final Function<File, FLH> tryLock;
+
+    /**
+     * @param cfg Ignite configuration.
+     * @param log Logger.
+     * @param consistentId Constent id.
+     * @param tryLock Lock function.
+     */
+    public PdsFolderResolver(
+        IgniteConfiguration cfg,
+        IgniteLogger log,
+        @Nullable Serializable consistentId,
+        Function<File, FLH> tryLock
+    ) {
+        this.cfg = cfg;
+        this.log = log;
+        this.consistentId = consistentId;
+        this.tryLock = tryLock;
+    }
+
+    /**
+     * Prepares compatible PDS folder settings. No locking is performed, consistent ID is not overridden.
+     *
+     * @param pstStoreBasePath DB storage base path or null if persistence is not enabled.
+     * @param consistentId compatibility consistent ID
+     * @return PDS folder settings compatible with previous versions.
+     */
+    private PdsFolderSettings<FLH> compatibleResolve(
+        @Nullable final File pstStoreBasePath,
+        @Nullable final Serializable consistentId) {
+
+        if (cfg.getConsistentId() != null) {
+            // compatible mode from configuration is used fot this case, no locking, no consitent id change
+            return new PdsFolderSettings<>(pstStoreBasePath, cfg.getConsistentId());
+        }
+
+        if (consistentId == null)
+            return new PdsFolderSettings<>(pstStoreBasePath, consistentId);
+
+        return null;
+    }
+
+    /**
+     * Creates new settings when we don't have cached one.
```

Comment:
We should fix the comment since a new consistent id generation was moved outside this method.

Reply:
Fixed. Thanks.